media/js/base/mozilla-monitor-button.js

```diff
@@ -7,30 +7,56 @@ if (typeof window.Mozilla === 'undefined') {
     window.Mozilla = {};
 }
 
-Mozilla.MonitorButton = (function() {
+(function() {
     'use strict';
 
-    var monitorButton = document.getElementById('fxa-monitor-submit');
+    var MonitorButton = {};
 
-    // fetch request
-    var supportsFetch = 'fetch' in window;
+    MonitorButton.init = function(buttonId){
 
-    if (!supportsFetch || !monitorButton) {
-        return;
-    }
+        // check for custom buttonId, used in case of multiple buttons.
+        if(!buttonId){
+            buttonId = 'fxa-monitor-submit';
+        }
 
-    var buttonURL = monitorButton.getAttribute('href');
-    var destURL = monitorButton.getAttribute('data-action') + 'metrics-flow';
+        var monitorButton = document.getElementById(buttonId);
 
-    fetch(destURL).then(function(resp) {
-        return resp.json();
-    }).then(function(r) {
-        // add retrieved flowBeginTime and flowId values to cta url
-        buttonURL += '&flowBeginTime=' + r.flowBeginTime;
-        buttonURL += '&flowId=' + r.flowId;
-        monitorButton.setAttribute('href', buttonURL);
-    }).catch(function() {
-        // silently fail, leaving flow_id and flow_begin_time as default empty value
-    });
+        // fetch request
+        var supportsFetch = 'fetch' in window;
+
+        if (!supportsFetch || !monitorButton) {
+            return;
+        }
+
+        var buttonURL = monitorButton.getAttribute('href');
+        // strip url to everything after `?`
+        var buttonURLParams = buttonURL.match(/\?(.*)/)[1];
+
+        var destURL = monitorButton.getAttribute('data-action') + 'metrics-flow';
+
+        // collect values from monitor button
+        var params = window._SearchParams.queryStringToObject(buttonURLParams);
+
+        // add required params to the token fetch request
+        destURL += '?entrypoint=' + params.entrypoint;
+        destURL += '&form_type=' + params.form_type;
+        destURL += '&utm_campaign=' + params.utm_campaign;
+        destURL += '&utm_content=' + params.utm_content;
```

Comment:
`utm_content` is coming through as `NaN` here, since it's not present in the monitor button URL. Because of this it surfaces as an empty parameter value in the final URL e.g. `utm_source=mozilla.org-firefox-accounts&utm_campaign=trailhead&utm_content=&form_type=email&`.

Because only `utm_source` is a required parameter, we should probably make both `utm_campaign` and `utm_content` optional here? e.g.

```javascript
if (params.utm_content) {
   destURL += '&utm_content=' + params.utm_content;
}
```